### DISCUSSION TITLE
Permanent Model Location

### DISCUSSION THREAD

**Discussion by khilller** [google/model-viewer#3859] [Q&A]
Hi All,
I was wondering if model-viewer has the ability to disable user interaction so that the model can be placed in one location and not moved. As an architect, this is beneficial when trying to show clients their designs on construction sites. Any and all suggestions would be greatly appreciated.

Thanks!

**Comment by elalish** [comment]
Can you be a little more specific? It sounds like you might want to just not add the [`camera-controls`](https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-cameraControls) attribute.

**Reply by khilller** [reply]
What I mean is once the asset is loaded I don't want to be able to move it around. I checked the code and I haven't used the `camera-controls` attribute and it still moves around, I have attached a link to the [video](https://share.icloud.com/photos/0b5quHOV2VNV90p3M0biSRF9w).

I was thinking maybe the ``disable-pan`` attribute could work. But no luck on that.

**Reply by elalish** [reply]
Hmm, your image is of AR mode - is that what you're asking about? Which AR mode are you using? If it's iOS, we have no control over QuickLook. Also, it doesn't look like you uploaded a video, but an image. You can just paste a video directly into Github, which is usually the easiest.

**Comment by khilller** [comment]
Yes to both, The asset is In AR mode on IOS. And sry about that, here's the video:

https://user-images.githubusercontent.com/29700551/194423810-9bc87786-2fe7-40af-92d8-723f1f707a65.mov

**Reply by elalish** [reply]
Yeah, we can't change Apple's app's behavior. But honestly, I don't think I'd want to anyway. Say the user needs to go somewhere? They'd have no choice but to leave the model behind. Also, it's no uncommon for models to get placed poorly on the first try.

**Reply by khilller** [reply]
Oh, that's unfortunate. And you are right but in some cases like a construction site, the user finds it hard to place. That's when I'm thinking it was beneficial for the model to get loaded when scanned. Do you think there could be a workaround for this? 

Also does the ``camera-control`` attribute work on android phones?

**Reply by elalish** [reply]
Yes, but it has nothing to do with AR. Camera control is automatic in AR by definition.

**Comment by khilller** [comment]
I thought as much, but didn't you mention in the first comment to possibly use that attribute? 
I'm trying to achieve this aspect of AR. What are your thoughts?

https://user-images.githubusercontent.com/29700551/194565119-5e9a3365-4420-47a5-b071-149185508bf4.mp4

Just looking at this I can tell that they may have used the Unity approach to create the app and place the model. I'm trying to make it a web-based approach with ``model-viewer`` sort of plug-and-play situation.

**Reply by elalish** [reply]
Wow, that's very cool! I'm not sure I'd trust vanilla ARCore for this kind of positional registration. Really you should be identifying features about this particular worksite to anchor things, rather than just looking for generic planes. I can't think of any way we could do this reliably without cloud anchors or something equivalent.

**Comment by khilller** [comment]
I think so too. If I were to use something as basic as a QR code, the model once loaded in the real world can be moved around. Which was what I asked the question in the beginning. Is there a way with ``model-viewer`` to place the model in the space (to a QR code for example) and not move at all?

**Reply by elalish** [reply]
We don't support QR-code tracking yet. But even if you do, that QR code should really remain in frame the whole time or the world will tend to drift a bit. For that particular use case on the construction site, I would be concerned about accuracy because of that.

**Comment by khilller** [comment]
Oh, ok. Thank you for your help!
